statsmodels/genmod/generalized_linear_model.py

```diff
@@ -714,17 +718,26 @@ def fit(self, start_params=None, maxiter=100, method='IRLS',
                                       tol=tol, scale=scale,
                                       full_output=full_output,
                                       disp=disp, cov_type=cov_type,
-                                      cov_kwds=cov_kwds, use_t=use_t, **kwargs)
+                                      cov_kwds=cov_kwds, use_t=use_t,
+                                      max_start_irls=max_start_irls,
+                                      **kwargs)
 
     def _fit_gradient(self, start_params=None, method="newton",
                       maxiter=100, tol=1e-8, full_output=True,
                       disp=True, scale=None, cov_type='nonrobust',
-                      cov_kwds=None, use_t=None, **kwargs):
+                      cov_kwds=None, use_t=None, max_start_irls=3,
+                      **kwargs):
         """
         Fits a generalized linear model for a given family iteratively
         using the scipy gradient optimizers.
         """
 
+        if max_start_irls > 0:
```

Comment:
I thought about that, but I think it's confusing if the method is "newton" and not "irls-newton".

Conceptually, the simple picture is that we use IRLS to get starting values. If we already have starting values, then we don't need IRLS. 
If I already have starting values that are pretty good, for example from a previous run, then I don't need IRLS and I don't want to have to check other keywords to see whether newton is really just newton in this model.

In this case it's just being picky and I think we could decide on either.
Currently we don't have anything like this, except maybe in tsa. The discrete model use the estimate of a simplified model to get the start_params if they are None. 
But we want to expand hybrid and sequence of optimizers to a general pattern, Skipper has a sequence optimization PR open, so we need to decide on some consistent rules.

My rule would be that, if I specify a method, then the start params are for that method. If I specify a hybrid method, then the start params are for the first optimizer in the hybrid.